src/sync/sync_manager.cpp

```diff
@@ -350,9 +349,10 @@ std::shared_ptr<SyncUser> SyncManager::get_user(const std::string& user_id,
         user->update_access_token(std::move(access_token));
         user->update_refresh_token(std::move(refresh_token));
 
-        if (user->state() == SyncUser::State::LoggedOut) {
-            user->set_state(SyncUser::State::LoggedIn);
-        }
+        user->set_state(SyncUser::State::LoggedIn);
+        if (!m_metadata_manager)
+            m_current_user = user;
+
```

Comment:
Why is there a need for this?

Reply:
I reasoned that we were forgetting to set the current user, but I just tested this and it is being done in a subsequent call to `App::get_profile()` elsewhere so indeed this is not necessary. I will remove it thanks!